Add RBAC enhancement documentation to NeuVector integration docs. (#4280)
rancher/docs: content/rancher/v2.6/en/neuvector-integration/rbac/_index.md

---

> title: Rancher + NeuVector RBAC

Reviewer: ```suggestion
title: Rancher and NeuVector RBAC
```

weight: 3
---

> This article is intended for users who need to provide access to the NeuVector app deployed via the Rancher App catalog with the Rancher chart. This will not work on deployments using the Partner chart.

Reviewer: ```suggestion
This article is intended for users who need to provide access to the NeuVector app deployed via the Rancher app catalog with the Rancher chart. This will not work on deployments using the partner chart.
```

> By default, a Rancher cluster admin, and global admin will be automatically mapped to be global admins within NeuVector. In order to map other personas, some access will need to be provided to the Rancher user/group depending on the desired access within NeuVector. Please note that adding the below permissions will not provide access to any kubernetes resources beyond what is already given by existing Rancher roles. With one exception being the neuvector service proxy.

Reviewer: ```suggestion
By default, a Rancher cluster admin and a global admin will automatically be mapped to be global admins within NeuVector. In order to map other personas, some access will need to be provided to the Rancher user/group depending on the desired access within NeuVector. Please note that adding the below permissions will not provide access to any Kubernetes resources beyond what is already given by existing Rancher roles, with one exception: the NeuVector service proxy.
```

> The following table lists the NeuVector role and the k8s RBAC from which it is derived. These rbac mappings need to be created within Rancher RBAC.

Reviewer: ```suggestion
The following table lists the NeuVector role and the Kubernetes RBAC from which it is derived. These RBAC mappings need to be created within Rancher RBAC.
```

> |NeuVector role|apiGroup |resources|verbs|comment|

Reviewer: ```suggestion
|NeuVector Role|API Group|Resources|Verbs|Comment|
```

|-----|-----|-----|-----|-----|
cluster admin|read-only.neuvector.api.io|*|*| clusterrole(with clusterrolebinding)|
cluster reader|read-only.neuvector.api.io|*|get| clusterrole(with clusterrolebinding)|
namespace admin|read-only.neuvector.api.io|*|*| clusterrole/role with rolebinding) via project|
namespace readonly|read-only.neuvector.api.io|*|get| clusterrole/role with rolebinding) via project|
n/a|neuvector.com|*|get|necessary along with any of the above for nav link to appear|

> ### Creating the rancher RBAC roles for cluster and project scope

Reviewer: ```suggestion
### Creating the Rancher RBAC Roles for Cluster and Project Scope
```

> _for users that are not global admins or cluster admins_

Reviewer: ```suggestion
>**Note:** This is applicable to users who are neither global admins nor cluster admins.
```

Three items are necessary for the mapped access:

> 1. Global, Cluster, or project level role based on the above table

Reviewer: ```suggestion
1. Global, cluster, or project level role based on the above table.
```

> 1. GET permissions on the neuvector.com CRDs

Reviewer: ```suggestion
1. GET permissions on the neuvector.com CRDs.
```

> 2. NeuVector Project level services/proxy permission. This is used for UI proxy via rancher.

Reviewer: ```suggestion
1. NeuVector project level services/proxy permission. This is used for UI proxy via Rancher.
```

> The first two items are highly dependent on your RBAC setup, but can be done with distinct NeuVector roles, or adding the permissions from the above tables to an existing set of custom roles. These can be given to users at Global, cluster, or project level.

Reviewer: ```suggestion
The first two items above are highly dependent on your RBAC setup but can be done with either distinct NeuVector roles or by adding the permissions from the above tables to an existing set of custom roles. These can be given to users at global, cluster, or project level.
```

See [Rancher Custom Roles]({{<baseurl>}}rancher/v2.6/en/admin-settings/rbac/default-custom-roles/) for more information.

> ### NeuVector Project Level UI Proxy

Reviewer: ```suggestion
### NeuVector Project Level UI Proxy

```

> _Necessary when a user does not have this permission already either via a global or cluster role_

Reviewer: ```suggestion
>**Note:** This is necessary when a user does not have this permission already either via a global or cluster role.
```

> 1. Create a project for NeuVector prior to installing from the App catalog, and install to this project. If install has already been done, create the project and move the namespace there.

Reviewer: ```suggestion
1. Create a project for NeuVector prior to installing from the App catalog and install to this project. If install has already been done, create the project and move the namespace there.
```

1. Create a project level role with services/proxy access as shown in the below examples.

> 1. For the user/group in question that will need to access NeuVector, assign the project UI Proxy role.

Reviewer: ```suggestion
1. For the user/group in question who will need to access NeuVector, assign the project UI proxy role.
```

> > **Warning**

Reviewer: ```suggestion
> **Warning:** Please be sure to scope this role to a NeuVector-only project. Otherwise, services/proxy access could be given to unintended workloads.
```

> > Please be sure to scope this role to a NeuVector only project, otherwise services/proxy access could be given to unintended workloads.

Reviewer: ```suggestion
```

### Examples

> #### Project level:

Reviewer: ```suggestion
#### Project Level:
```

![Project Admin]({{<baseurl>}}/img/rancher/neuvector-project-admin.png)
![Project Read-Only]({{<baseurl>}}/img/rancher/neuvector-project-ro.png)

> ![Project UI Proxy]({{<baseurl>}}/img/rancher/neuvector-proxy-role.png)

Reviewer: ```suggestion
![Project UI Proxy]({{<baseurl>}}/img/rancher/neuvector-proxy-role.png)

```

> #### Cluster level:

Reviewer: ```suggestion
#### Cluster Level:
```

![Cluster Admin]({{<baseurl>}}/img/rancher/neuvector-cluster-admin.png)
![Cluster Read-Only]({{<baseurl>}}/img/rancher/neuvector-cluster-ro.png)
 

> #### Project UI proxy permission:

Reviewer: ```suggestion
#### Project UI Proxy Permission:
```

---
title: Rancher and NeuVector RBAC
weight: 3
---

This article is intended for users who need to provide access to the NeuVector app deployed via the Rancher app catalog with the Rancher chart. This will not work on deployments using the partner chart.

> By default, a Rancher cluster admin and a global admin will automatically be mapped to be global admins within NeuVector. In order to map other personas, some access will need to be provided to the Rancher user/group depending on the desired access within NeuVector. Please note that adding the below permissions will not provide access to any Kubernetes resources beyond what is already given by existing Rancher roles, with one exception: the NeuVector service proxy.

Reviewer 2: > Please note that adding the below permissions will not provide access to any Kubernetes resources beyond what is already given by existing Rancher roles, with one exception: the NeuVector service proxy.

I still don't think this is accurate. If I'm a project member who only has access to my single custom project, and then you give me NV admin permissions, I can now see many things I couldn't see before. I just have to click on Network activity and now I have indirect `GET`/`LIST` access on all sorts of things: cattle-cluster-agent, fleet-agent, node information, ingress information, traffic flow between them etc... I don't have the raw k8s RBAC to get that info directly from the API server, but I've just used the NV service-account to proxy that information for me. Once you get into policy stuff you can go beyond Get/List and into Create/Update.
